### PR TITLE
[FLINK-5871] [cep] Enforce uniqueness of pattern names in CEP.

### PR DESCRIPTION
**Pull request by kl0u**
Description of the problem can be found in the related JIRA:

https://issues.apache.org/jira/browse/FLINK-5871

### CODE REVIEW

File: flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/compiler/NFACompilerTest.java [hunk 1]
```diff
@@ -42,6 +45,43 @@
...
 
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none();
+
+	@Test
+	public void testNFACompilerUniquePatternName() {
+
+		// adjust the rule
+		expectedException.expect(MalformedPatternException.class);
+		expectedException.expectMessage("Duplicate pattern name: start. Pattern names must be unique.");
```
Comment (tillrohrmann) [inline flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/compiler/NFACompilerTest.java]: Matching on exception messages is fragile

File: flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/compiler/NFACompilerTest.java [hunk 2]
```diff
@@ -42,6 +45,43 @@
...
+				return value.getVolume() > 10.0;
+			}
+		}).followedBy("start").subtype(SubEvent.class).where(new FilterFunction<SubEvent>() {
+			private static final long serialVersionUID = -1413973420880409702L;
+
+			@Override
+			public boolean filter(SubEvent value) throws Exception {
+				return value.getVolume() > 11.0;
+			}
+		}).within(Time.milliseconds(10));
```
Comment (tillrohrmann) [inline flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/compiler/NFACompilerTest.java]: Test pattern could be much more concise for the test case.

File: flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/compiler/NFACompilerTest.java (hunk 1 above)
Comment (kl0u) [inline flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/compiler/NFACompilerTest.java]: I match on both type and message. Do you suggest to remove the message matching part or you have a better way to do so?

File: flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/compiler/NFACompilerTest.java (hunk 2 above)
Comment (kl0u) [inline flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/compiler/NFACompilerTest.java]: Thanks for the review @tillrohrmann . I just had a question about one of your comments.

**Comment by kl0u** [comment]
Thanks for the review @tillrohrmann ! I just had a question about one of the comments.

**Pull request merged**